### PR TITLE
Slot Approvals implementation

### LINKED ISSUE #4: Implementation of Interface IERC3525SlotApprovable
**Description**
---
Implement the functions and events of the Interface `IERC3525SlotApprovable` of the [EIP-3525](https://eips.ethereum.org/EIPS/eip-3525). 
The `IERC3525SlotApprovable` interface, slot’s enumeration extension is OPTIONAL.

The list of functions are :
- `setApprovalForSlot`
- `isApprovedForSlot`

The list of events are :
- `ApprovalForSlot`

**Rules**
---

approving rules:

This EIP provides four kinds of approving functions indicating different levels of approvals, which can be described as full level approval, slot level approval, token ID level approval as well as value level approval.

- `setApprovalForAll`, compatible with EIP-721, SHOULD indicate the full level of approval, which means that the authorized operators are capable of managing all the tokens, including their values, owned by the owner.
- `setApprovalForSlot` (optional) SHOULD indicate the slot level of approval, which means that the authorized operators are capable of managing all the tokens with the specified slot, including their values, owned by the owner.
- The token ID level `approve` function, compatible with EIP-721, SHOULD indicate that the authorized operator is capable of managing only the specified token ID, including its value, owned by the owner.
- The value level `approve` function, SHOULD indicate that the authorized operator is capable of managing the specified maximum value of the specified token owned by the owner.
- For any approving function, the caller MUST be the owner or has been approved with a higher level of authority.

**Acceptance Criteria**
---
- [x] Implement the contents of each functions and events 
- [x] Implement unit tests for each functions
- [x] Implement integration tests for each functions
- [ ] If needed, write the documentation associated with the functions and events exposed

### PR DESCRIPTION
**Pull request by tekkac**
Closes #4

### CODE REVIEW

**Review by cloudvenger — DISMISSED** [review]
LGFM

File: src/carbonable/erc3525/extensions/slotapprovable/library.cairo [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        with_attr error_message("ERC3525SlotApprovable: approved is not a Cairo boolean") {
+            assert 0 = approved * (1 - approved);
+        }
+
+        ERC3525SlotApprovable_slot_approvals.write(owner, slot, operator, approved);
+        ApprovalForSlot.emit(owner, slot, operator, approved);
+        return ();
+    }
+
+    @external
```
Comment (bal7hazar) [inline src/carbonable/erc3525/extensions/slotapprovable/library.cairo]: Not sure if it is a mistake or not, but I'm surprised to see decorators in library

**Review by bal7hazar — DISMISSED** [review]
Nice to see that you left 3525 naming to a more explicit one

File: src/carbonable/erc3525/extensions/slotapprovable/library.cairo (hunk 1 above)
Comment (tekkac) [reply src/carbonable/erc3525/extensions/slotapprovable/library.cairo]: Good catch, copypasta typos. Done

**Review by bal7hazar — APPROVED** [review]
LGTM

**Pull request merged**